Phoenix example (#45)
secrethub/examples: application/phoenix/README.md

> This Phoenix example checks if the environment variables `DEMO_USERNAME` and `DEMO_PASSWORD` have been set. If that's the case, you'll receive a personalized greeting message on http://localhost:4000, otherwise you will not be recognized.

Reviewer: ```suggestion
This Phoenix example checks if the environment variables `DEMO_USERNAME` and `DEMO_PASSWORD` have been set. If that's the case, you'll receive a personalized greeting message on http://localhost:4000, otherwise you will get a "Hello stranger" message.
```

I think a more specific sentence would be better here.

Author: fixed

> This Phoenix example checks if the environment variables `DEMO_USERNAME` and `DEMO_PASSWORD` have been set. If that's the case, you'll receive a personalized greeting message on http://localhost:4000, otherwise you will get an error message.

Reviewer 2: We now have a graphic to use as a heading: https://secrethub.io/img/integrations/phoenix/github-banner.png. See how the other examples do it.

Author: Alright, it's added. I will do the same for all the other guides.

> This Phoenix example checks if the environment variables `DEMO_USERNAME` and `DEMO_PASSWORD` have been set. If that's the case, you'll receive a personalized greeting message on http://localhost:4000, otherwise you will get a "Hello stranger" message.

Reviewer 2: This "hello stranger" thing is new to me, I'd prefer to change that so it's consistent with the other examples:
 - [Ruby on Rails](https://github.com/secrethub/examples/blob/c8e2b98c03b8a155ed2e0afb620fa71f6afe8b67/application/rails/application_controller.rb)
 - [Django](https://github.com/secrethub/examples/blob/c8e2b98c03b8a155ed2e0afb620fa71f6afe8b67/application/django/views.py)
 - [Flask](https://github.com/secrethub/examples/blob/c8e2b98c03b8a155ed2e0afb620fa71f6afe8b67/application/flask/app.py)

Author: After some investigation, this proves to be a little cumbersome to make with my actual skillset in Elixir/Phoenix. I will try to revisit this and to fix it accordingly as soon as I'll have some time to read more on it.

Author: This has been fixed. I've tried it out and it now works as expected.

## Prerequisites
1. [Docker](https://docs.docker.com/install/) installed and running
1. [SecretHub](https://secrethub.io/docs/start/getting-started/#install) installed
1. A SecretHub repo that contains a `username` and `password` secret. To create it, run `secrethub demo init`.

## Running the example

Set the SecretHub username in an environment variable
```
export SECRETHUB_USERNAME=<your-username>
```

Create a service account for the demo repo
```
secrethub service init --description demo_service \
--permission read --file demo_service.cred ${SECRETHUB_USERNAME}/demo
```

Build the Phoenix docker demo
```
docker build . -t phoenix-secrethub-demo
```

Run the docker demo with the secrets in the environment variables
```

> docker run -p 8080:8080 \

Reviewer: https://github.com/secrethub/examples/blob/d66fc06d6af69805b8ce8403403b1fe97287d20c/application/phoenix/Dockerfile#L15

In the Dockerfile you exposed port `4000` but here you map port `8080` to port `8080`. This is a mistake, right?

Author: Yeppers, I'm fixing this now.
